src/services/signatureHelp.ts

```diff
@@ -128,10 +129,40 @@ namespace ts.SignatureHelp {
     }
     export function getArgumentInfoForCompletions(node: Node, position: number, sourceFile: SourceFile): ArgumentInfoForCompletions | undefined {
         const info = getImmediatelyContainingArgumentInfo(node, position, sourceFile);
-        return !info || info.kind === ArgumentListKind.TypeArguments || info.invocation.kind === InvocationKind.TypeArgs ? undefined
+        return !info || info.isTypeParameterList || info.invocation.kind !== InvocationKind.Call ? undefined
             : { invocation: info.invocation.node, argumentCount: info.argumentCount, argumentIndex: info.argumentIndex };
     }
 
+    function getArgumentOrParameterListInfo(node: Node, sourceFile: SourceFile): { readonly list: Node, readonly argumentIndex: number, readonly argumentCount: number, readonly argumentsSpan: TextSpan } | undefined {
+        const info = getArgumentOrParameterListAndIndex(node, sourceFile);

```

Comment:
Why not inline this function, as we would avoid unnecessary object creation and destructuring then.

Reply:
It's called in two places -- inlining it twice would increase code size which could also hurt performance..